src/core/meshcop/dataset_updater.cpp

```diff
@@ -95,6 +95,7 @@ Error DatasetUpdater::RequestUpdate(Dataset &aDataset, UpdaterCallback aCallback
     else
     {
         pendingTimestamp.Clear();
+        pendingTimestamp.SetSeconds(1);
```

Comment:
The reason we need `SetSeconds(1)` here is that `AppendPendingTimestampTlv()` skips appending the TLV if the timestamp's `GetSecond()` is zero. Previously, `AdvanceRandomTicks()` always incremented the seconds, which happened to make it work correctly.

https://github.com/openthread/openthread/blob/b2f17f865d685d62dd1bcca89b39d3d77b5425f5/src/core/thread/mle.cpp#L4748-L4758

It's unclear to me whether the check `GetSecond() != 0` is necessary (it exists only in `AppendPendingTimestampTlv()` and not in `AppendActiveTimestampTlv()`). We could potentially remove this in a future PR. @jwhui, thoughts?

Reply:
Removing the check SGTM. Thanks.

Reply:
Thanks. Submitted https://github.com/openthread/openthread/pull/10327.